### ISSUE TITLE
Enhance Feature: smooth_power_on

### ISSUE THREAD

**Issue by davidclemens** [xaviml/controllerx#15]
I love the smooth_power_on feature.
It would be great if you could extent the functionality to all 4 outer buttons on the E1524 with the following or similar mapping if pushed while the light is off:

* brightness_up -> turn on @ 255 brightness (with previous color/color_temp)
* brightness_down -> turn on @ 1 brightness (with previous color/color_temp)
* left -> turn on @ 153 mireds (with previous brightness)
* right -> turn on @ 500 mireds (with previous brightness)

That would be great. Maybe you could define new actions from this like "brightness_up_from_off".
It would be even greater, if this mapping was also user configurable. For example like:

brightness_up_from_off:
  brightness: 1
  mireds: 327
brightness_down_from_off:
  brightness: 255
  mireds: previous
arrow_left_from_off:
  brightness: previous
  mireds: 153

and so on.

This would maybe tie in with issue #12.

Thanks a lot!

**Comment by xaviml** [comment]
Hi @davidclemens 

Thanks for the idea. I am currently refactoring the project, so I will consider this once I finish with it and I might consider it to impement this with a CustomController (#12) as you said.

**Comment by xaviml** [comment]
Hi @davidclemens,

I recently released a beta version v2.3.0b1 with the new functionality for custom controllers. This new feature allows users to configure the behaviour for the controllers through yaml configuration. From now on, you will be able to do what you wanted with the power of custom controllers and [callback constraints](https://www.home-assistant.io/docs/ecosystem/appdaemon/api/#callback-constraints). I prepared a configuration that does what you want. It can also be found in the [examples page](https://xaviml.github.io/controllerx/start/examples).

```yaml
livingroom_light_on:
  module: controllerx
  class: E1810Controller
  controller: sensor.livingroom_controller_action
  integration: z2m
  light: light.livingroom
  constrain_input_boolean: light.livingroom,on
  
livingroom_light_off:
  module: controllerx
  class: CustomLightController
  controller: sensor.livingroom_controller_action
  integration: z2m
  light: light.livingroom
  mapping:
    toggle: toggle
    brightness_up_click: on_full_brightness
    brightness_down_click: on_min_brightness
    brightness_up_hold: on_full_brightness
    brightness_down_hold: on_min_brightness
    arrow_right_click: on_full_color_temp
    arrow_left_click: on_min_color_temp
    arrow_right_hold: on_full_color_temp
    arrow_left_hold: on_min_color_temp
  constrain_input_boolean: light.livingroom,off
```

If you want to know more about custom controllers, you can check it out in [here](https://xaviml.github.io/controllerx/how-to/custom-controllers).

**Comment by davidclemens** [comment]
Thank you very much! Great work @xaviml !
